### PR TITLE
WIP: Structopt 0.2 without macro re-export in quicli

### PR DESCRIPTION
**Pull request by killercup**
It doesn't look like I can get it to re-export the proc macro like I can for the proc-macro _derive crate.

Since we already tell people to add structopt to their Cargo.toml, it's might not look like a big step to make them add `#[macro_use] extern crate structopt;` to their files, too. But still, this sadly is worse than what we have right now.

cf. https://github.com/TeXitoi/structopt/pull/50#issuecomment-363039922

cc @TeXitoi

### CODE REVIEW

**Comment by TeXitoi** [comment]
Aybe you can export directly from structopt_derive as before.

**Comment by TeXitoi** [comment]
You still can use structopt_derive as before, that's just not the easiest way for users. But in tour case that's just fine.

**Comment by CAD97** [comment]
Does `#[macro_reexport]` change anything?

File: src/lib.rs [hunk 1]
```diff
@@ -26,10 +25,6 @@ mod main_macro;
 mod reexports {
     #[doc(hidden)] pub use serde_derive::*;
 
-    #[doc(hidden)] pub use structopt_derive::*;
-    #[doc(hidden)] pub mod structopt {
-        pub use ::structopt::*;
-    }
```
Comment (TeXitoi) [inline src/lib.rs]: I think you still need something like that. At worst, it should work as before if you don't remove all the section.

Note that the custom derive will generate something containing ```::structopt::StructOpt``` thus tour reexport dance should include something like that.

**Comment by TeXitoi** [comment]
I think there is doc in quicli that must be updated for stoping special casing of u64 for v0.2, as I've seen that the main macro can use an integer parameter to set the default log level.

File: Cargo.lock [hunk 2]
```diff
@@ -68,13 +68,13 @@ dependencies = [
 
 [[package]]
 name = "clap"
-version = "2.29.2"
+version = "2.29.4"
```
[the remark this answers is not in the document]
Comment (killercup) [reply Cargo.lock]: I actually kind of want to have a lock file to know which versions worked but also have a CI build that always uses the newest (or the oldest compatible) versions. I'll need to think about this a bit more.

**Comment by killercup** [comment]
Ah, I'm stupid and it was pretty late. You are totally right, let me open another PR.

**Pull request closed without merging**